lib/common/patchset_display.c

```diff
@@ -132,27 +147,104 @@ xml_show_patchset_v1(pcmk__output_t *out, const xmlNode *patchset)
     removed = find_xml_node(patchset, "diff-removed", FALSE);
     for (child = pcmk__xml_first_child(removed); child != NULL;
          child = pcmk__xml_next(child)) {
-        xml_show_patchset_v1_recursive(out, "- ", child, 0,
-                                       options|pcmk__xml_fmt_diff_minus);
+        int temp_rc = xml_show_patchset_v1_recursive(out, "- ", child, 0,
+                                                     options
+                                                     |pcmk__xml_fmt_diff_minus);
+        rc = pcmk__output_select_rc(rc, temp_rc);
+
         if (is_first) {
             is_first = false;
         } else {
-            out->info(out, " --- ");
+            rc = pcmk__output_select_rc(rc, out->info(out, " --- "));
         }
     }
 
     is_first = true;
     added = find_xml_node(patchset, "diff-added", FALSE);
     for (child = pcmk__xml_first_child(added); child != NULL;
          child = pcmk__xml_next(child)) {
-        xml_show_patchset_v1_recursive(out, "+ ", child, 0,
-                                       options|pcmk__xml_fmt_diff_plus);
+        int temp_rc = xml_show_patchset_v1_recursive(out, "+ ", child, 0,
+                                                     options
+                                                     |pcmk__xml_fmt_diff_plus);
+        rc = pcmk__output_select_rc(rc, temp_rc);
+
         if (is_first) {
             is_first = false;
         } else {
-            out->info(out, " +++ ");
+            rc = pcmk__output_select_rc(rc, out->info(out, " +++ "));
         }
     }
+
+    return rc;
+}
+
+/*!
+ * \internal
+ * \brief Output an XML patchset (format 1)
+ *
+ * This function parses an XML patchset (an \p XML_ATTR_DIFF element and its
+ * children) into a user-friendly combined diff output.
+ *
+ * \param[in,out] out       Output object
+ * \param[in]     patchset  XML patchset to output
+ *
+ * \return Standard Pacemaker return code
+ */
+PCMK__OUTPUT_ARGS("xml-patchset-v1", "xmlNode *")
+static int
+xml_patchset_v1_default(pcmk__output_t *out, va_list args)
+{
+    xmlNode *patchset = va_arg(args, xmlNode *);
```

Comment:
This can be const for default and log. It just has to be non-const for XML due to doc stuff.